### PR TITLE
support for sles-12 and redhat7

### PR DESCRIPTION
**Pull request by pguibert6WIND**
hello @maheshloni ,
i would like you to review this pull request.

since i replace facter tool with lsb_release.

regarding the usage of facter, it is a mess to install it on sles-12.
that is why we want to propose using lsb_release instead.

Can I ask you to check if lsb_release is installed on other distribs ?
Thanks

### CODE REVIEW

**Comment by maheshloni** [comment]
Hi Philippe,
lsb_release  command  doesn't work in Centos7. More over CSIT accepts only using facter command (standard command) 

Note: CSIT  suggested standard command to fetch OS and its version from various flavors of operating system instead of using lsb_release or uname -a 

I think its better to add one more option for sles12 : zypper install facter  by appending in the below line #60 :
         apt-get install -y facter || yum -y install facter || < sles12 command to install facter>   ;

 and for redhat os : yum command will work for installing the facter .
Can you please check on sles12 whether zypper install facter is works or not  . If you know proper syntax to install the facter command in sles12 , then its better to add it.

**Comment by pguibert6WIND** [comment]
Hi @maheshloni ,
a new rework of the script is available here.
please review.

**Comment by pguibert6WIND** [comment]
Hi @maheshloni , please review the pull request.
some update has been done.

**Comment by pguibert6WIND** [comment]
use following command to test:

./pkgsrc/dev_compile_script.sh -p -d -b -v 5

**Comment by xiaofeng6WIND** [comment]
Hi @maheshloni , please review new patches:
- last commited id and distribution is added into package name
- support log rotation
- support systemd for Redhat

Thanks.

**Review by maheshloni — APPROVED** [review]

**Review by maheshloni — APPROVED** [review]
will test this

**Comment by xiaofeng6WIND** [comment]
Hi @pguibert6WIND @maheshloni  , Please review the patches:
1. pkgsrc: package c-capnproto
2. pkgsrc: package thrift
3. pkgsrc: package zmq

File: pkgsrc/packaging.sh [hunk 1]
```diff
@@ -304,6 +304,93 @@ quagga_deb_bin_control () {
...
+    pushd $INST_BIN_DIR
+    if [ $PACKAGE_DEB = "y" ]; then
+       find . \! -type d | cpio -o -H ustar -R 0:0 | tar -C $DEB_BIN_DIR -x
+    fi
+    popd
+}
+
+ccapnproto_rpm_bin_spec () {
+    echo "Name: c-capnproto" >> $RPM_SPEC_FILE
+    if [ -z "$COMMITID" ]; then
```
Comment (pguibert6WIND) [inline pkgsrc/packaging.sh:317]: I have one remark:
- c-capnproto is version 1.0.2. And it almost never changes.
I would say that the packging name should not change. => no commit ID.

File: pkgsrc/packaging.sh [hunk 2]
```diff
@@ -391,6 +391,111 @@ ccapnproto_deb_bin_control () {
...
+    pushd $INST_BIN_DIR/../bin
+    if [ $PACKAGE_DEB = "y" ]; then
+       find . \! -type d | cpio -o -H ustar -R 0:0 | tar -C $DEB_BIN_DIR -x
+    fi
+    popd
+}
+
+thrift_rpm_bin_spec () {
+    echo "Name: thrift" >> $RPM_SPEC_FILE
+    if [ -z "$COMMITID" ]; then
```
Comment (pguibert6WIND) [inline pkgsrc/packaging.sh:413]: I don' think it is needed to package thrift under commit ID naming specific too

**Review by pguibert6WIND — COMMENTED** [review]
some comments on the packaging name for c-capnproto, zmq, and thrift.

**Review by maheshloni — COMMENTED** [review]
will test and update

**Pull request merged**